amivapi/tests/blacklist/test_emails.py

```diff
@@ -0,0 +1,131 @@
+                      status_code=201)
+
+        # Look for sent out mail
+        mail = self.app.test_mails[0]
+        self.assertEqual(mail['receivers'], 'bla@bla.bl')
+        expected_text = (
+            "You have been blacklisted for the following reason:\n\nTest"
+            "\n\nThis means that you cannot register for any new amiv events! "
+            "To get removed from the blacklist, you have to pay 5.50 CHF\n\n"
+            "If you think that this is an error, don't hesitate to contact "
+            "bouncer@amiv.ethz.ch\n\nBest Regards,\nAMIV"
+        )
+        self.assertEqual(mail['text'], expected_text)
+
+    def test_receive_email_on_new_entry_wo_price(self):
+        """Test if a user receives an email if he is added to the blacklist."""
+        # Create a user
+        self.load_fixture({'users': [{'_id': 24 * "0", 'email': "bla@bla.bl"}]})
+        data = {
+            'user': 24 * "0",
+            'reason': "Test",
+            'start_time': "2019-03-29T00:00:00Z",
+            'end_time': "2019-03-30T00:00:00Z",
+        }
+        self.api.post("/blacklist", data=data, token=self.get_root_token(),
+                      status_code=201)
+
+        # Check mail
+        mail = self.app.test_mails[0]
+        self.assertEqual(mail['receivers'], 'bla@bla.bl')
+        expected_text = (
+            "You have been blacklisted for the following reason:\n\nTest"
+            "\n\nThis means that you cannot register for any new amiv "
+            "events!\n\nIf you think that this is an error, don't hesitate to "
+            "contact bouncer@amiv.ethz.ch\n\nBest Regards,\nAMIV"
+        )
+        self.assertEqual(mail['text'], expected_text)
+
+    def test_receive_email_on_patch(self):
+        """Test if a user receives a mail if one of his entries is resolved."""
+        user_id = 24 * '0'
+        blacklist_id = 24 * '1'
+
+        # Create user and blacklist entry
+        self.load_fixture({'users': [{'_id': user_id, 'email': "bla@bla.bl"}]})
+        r = self.load_fixture({
+                'blacklist': [{
+                    '_id': blacklist_id,
+                    'user': user_id,
+                    'reason': "Test1",
+                    'end_time': datetime(2017, 1, 1)}]
+            })
+
+        etag = r[0]['_etag']
+
+        patch = {'reason': "Test2"}
+        header = {'If-Match': etag}
+        with freeze_time(datetime(2017, 1, 2)):
+            r = self.api.patch("/blacklist/%s" % blacklist_id, data=patch,
+                               headers=header, token=self.get_root_token(),
+                               status_code=200)
+
+        # Check mail
+        mail = self.app.test_mails[0]
+        self.assertEqual(mail['receivers'], 'bla@bla.bl')
+        expected_text = (
+            "Congratulations, your blacklist entry with the following reason "
+            "has been removed:\n\nTest1\n\nBest Regards,\nAMIV"
+        )
+        self.assertEqual(mail['text'], expected_text)
```

Comment:
After modifying the format messages, I extended the tests to make sure that the correct text ist sent.

Through this, I discovered that after a patch, the wrong email seems to be sent. The text should be that the blacklist entry is removed, as the date is in the past, but the sent email says that the user is now on the blacklist.